src/main/java/dk/xam/jbang/Main.java

```diff
@@ -260,6 +260,42 @@ public Integer call() throws IOException {
 		}
 	}
 
+	static boolean checkVersions() {
+		String version = System.getProperty("java.version");
+		if (version.startsWith("1.")) {
+			version = version.substring(2, 3);
+		} else {
+			int dot = version.indexOf(".");
+			if (dot != -1) {
+				version = version.substring(0, dot);
+			}
+		}
+		int javaVersion = Integer.parseInt(version);
+		int javacVersion = 0;
+		try {
+			Process p = Runtime.getRuntime().exec("javac -version");
+			StringBuilder output = new StringBuilder();
+			BufferedReader reader = new BufferedReader(new InputStreamReader(p.getInputStream()));
+			String line;
+			while ((line = reader.readLine()) != null) {
+				output.append(line + "\n");
+			}
+			String[] s = output.toString().split(" ");
+			if (s[1].startsWith("1.")) {
+				s[1] = s[1].substring(2, 3);
+			} else {
+				int dot = s[1].indexOf(".");
+				if (dot != -1) {
+					s[1] = s[1].substring(0, dot);
+				}
+			}
+			javacVersion = Integer.parseInt(s[1]);
+		} catch (IOException e) {
+			e.printStackTrace();
```

Comment:
we should not just print stacktraces - we use info/warn with a e as argument so printing is automatically handled.

Reply:
I have changed this part to print e using Util.warnMsg.

Reply:
you haven't pushed that code to this PR yet have you ?

Reply:
No I have not I thought I'll implement the told changes and then push it.